libs/hdf-converters/src/neuvector-mapper.ts

```diff
@@ -0,0 +1,262 @@
+    tty: true
+    command: tygo generate
+```
+
+// `docker compose` command to regenerate Tygo types: `docker compose up --build go2ts`. Output file will be `./tygo-output/neuvector-generated-types.ts`. 
+Some of the generated types are copied into and defined in `heimdall2/libs/hdf-converters/types/neuvector-types.ts`. The currently-used generated types begin with `REST`, and some of the generated types are tweaked such that when they have a field beginning with `REST`, the field is intersected instead. For example: 
+```
+type RESTScanRepoReport = {
+  verdict?: string;
+  image_id: string;
+  registry: string;
+  repository: string;
+  tag: string;
+  digest: string;
+  size: number;
+  author: string;
+  base_os: string;
+  created_at: string;
+  cvedb_version: string;
+  cvedb_create_time: string;
+  layers: RESTScanLayer[];
+} & RESTScanReport;
+```
+was tweaked from the generated
+```
+export interface RESTScanRepoReport {
+  verdict?: string;
+  image_id: string;
+  registry: string;
+  repository: string;
+  tag: string;
+  digest: string;
+  size: number;
+  author: string;
+  base_os: string;
+  created_at: string;
+  cvedb_version: string;
+  cvedb_create_time: string;
+  layers: (RESTScanLayer | undefined)[];
+  RESTScanReport: RESTScanReport;
+}
+```
+. In the original Golang RESTScanRepoReport struct at https://github.com/neuvector/neuvector/blob/15496f08f7c445acd4901105fa9e73637b72cdf7/controller/api/apis.go#L2444-L2459, RESTScanReport is composed within RESTScanRepoReport. In Golang, this allows RESTScanRepoReport to use RESTScanReport's members. Tygo embeds RESTScanReport inside of RESTScanRepoReport, but when comparing the actual NeuVector JSON output to the struct definitions, RESTScanReport is not a field of the output's `report` field. In TypeScript, the equivalent of a type accessing another type's members is type intersection. Additionally, NeuVectorScanJson is handcrafted, as its equivalent type doesn't exist in `neuvector-generated-types.ts`.
+*/
+
+const CWE_NIST_MAPPING = new CweNistMapping();
+
+function cweTags(description: string): string[] | undefined {
+  const regex = /CWE-\d{3}/g;
+  return description.match(regex) ?? undefined;
+}
+
+function nistTags(cweTags: string[] | undefined): string[] {
+  const identifiers = cweTags?.map((tag: string) => tag.slice(-3)) ?? [];
+  return CWE_NIST_MAPPING.nistFilter(
+    identifiers,
+    DEFAULT_UPDATE_REMEDIATION_NIST_TAGS
+  );
+}
+
+function cveIdMatches(cveName: string): (value: RESTModuleCve) => boolean {
+  return (cve: RESTModuleCve) => cve.name === cveName;
+}
+
+export class NeuVectorMapper extends BaseConverter {
+  withRaw: boolean;
+  rawData: NeuVectorScanJson;
+
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {
+      duration: null
```

Comment:
please get rid of all the unused attributes such as the ones that get assigned null